### PR TITLE
Improve TypeScript generation

### PR DESCRIPTION
**Pull request by christian-bromann**
## Proposed changes

We had a helper file that would treat certain commands different. It seems that we actually don't need all types in there. This patch cleans some of the stuff up and enhances some typings.

## Types of changes

[//]: # (What types of changes does your code introduce to WebdriverIO?)
[//]: # (_Put an `x` in the boxes that apply_)

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Documentation update

## Checklist

[//]: # (_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._)

- [x] I have read the [CONTRIBUTING](https://github.com/webdriverio/webdriverio/blob/master/CONTRIBUTING.md) doc
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)
- [x] I have added proper type definitions for new commands (if appropriate)

### CODE REVIEW

File: packages/wdio-sync/webdriverio-core.d.ts [hunk 1]
```diff
@@ -680,9 +702,11 @@ declare namespace WebdriverIO {
 
         /**
          * Retrieve a [cookie](https://w3c.github.io/webdriver/webdriver-spec.html#cookies)
+         * visible to the current page. You can query a specific cookie by providing the cookie name or
+         * retrieve all.
          */
         getCookies(
```
Comment (erwinheitzman) [inline packages/wdio-sync/webdriverio-core.d.ts]: I am missing the deleteCookies command

Comment (christian-bromann) [reply packages/wdio-sync/webdriverio-core.d.ts]: right here: https://github.com/webdriverio/webdriverio/blob/master/packages/wdio-sync/webdriverio-core.d.ts#L674-L679

File: packages/webdriverio/src/commands/browser/getCookies.js [hunk 2]
```diff
@@ -25,8 +25,8 @@
  * </example>
  *
  * @alias browser.getCookies
- * @param {String[]=} names  names of requested cookies
- * @return {Object[]}        requested cookies if existing
+ * @param {String[]=|String=} names  names of requested cookies (if omitted, all cookies will be returned)
```
Comment (erwinheitzman) [inline packages/webdriverio/src/commands/browser/getCookies.js]: deleteCookies might need the same changes

File: packages/webdriverio/webdriverio-core-v5.d.ts [hunk 3]
```diff
@@ -680,9 +702,11 @@ declare namespace WebdriverIO {
 
         /**
          * Retrieve a [cookie](https://w3c.github.io/webdriver/webdriver-spec.html#cookies)
+         * visible to the current page. You can query a specific cookie by providing the cookie name or
+         * retrieve all.
          */
         getCookies(
-            names?: string[]
+            names?: string[] | string
```
Comment (erwinheitzman) [inline packages/webdriverio/webdriverio-core-v5.d.ts]: missing deleteCookies

File: packages/webdriverio/webdriverio-core.d.ts [hunk 4]
```diff
@@ -680,9 +702,11 @@ declare namespace WebdriverIO {
 
         /**
          * Retrieve a [cookie](https://w3c.github.io/webdriver/webdriver-spec.html#cookies)
+         * visible to the current page. You can query a specific cookie by providing the cookie name or
+         * retrieve all.
          */
         getCookies(
-            names?: string[]
+            names?: string[] | string
```
Comment (erwinheitzman) [inline packages/webdriverio/webdriverio-core.d.ts]: missing deleteCookies

File: packages/webdriverio/src/commands/browser/getCookies.js (hunk 2 above)
Comment (christian-bromann) [reply packages/webdriverio/src/commands/browser/getCookies.js]: true, adjusted it

File: packages/webdriverio/webdriverio-core-v5.d.ts (hunk 3 above)
Comment (christian-bromann) [reply packages/webdriverio/webdriverio-core-v5.d.ts]: `deleteCookies` is defined in this file

**Comment by christian-bromann** [comment]
@erwinheitzman deleteCookies is defined in every file

**Comment by mgrybyk** [comment]
I think we should delete `packages/webdriverio/webdriverio-core-v5.d.ts` in v6 if it still exists

**Comment by christian-bromann** [comment]
removed the typings for v5

File: packages/webdriverio/webdriverio-core.d.ts [hunk 5]
```diff
@@ -816,6 +847,16 @@ declare namespace WebdriverIO {
...
             url?: string
         ): Promise<void>;
+
+        /**
+         * This wait command is your universal weapon if you want to wait on something. It expects a condition
+         * and waits until that condition is fulfilled with a truthy value. If you use the WDIO testrunner the
+         * commands within the condition are getting executed synchronously like in your test.
+         */
+        waitUntil(
+            condition: Function,
```
Comment (mgrybyk) [inline packages/webdriverio/webdriverio-core.d.ts]: It should be `condition: () => Promise<boolean>,` for async version

File: packages/wdio-sync/webdriverio-core.d.ts [hunk 6]
```diff
@@ -816,6 +847,16 @@ declare namespace WebdriverIO {
...
             url?: string
         ): void;
+
+        /**
+         * This wait command is your universal weapon if you want to wait on something. It expects a condition
+         * and waits until that condition is fulfilled with a truthy value. If you use the WDIO testrunner the
+         * commands within the condition are getting executed synchronously like in your test.
+         */
+        waitUntil(
+            condition: Function,
```
Comment (mgrybyk) [inline packages/wdio-sync/webdriverio-core.d.ts]: Do you think we can change it to `condition: () => boolean,` somehow?

Comment (christian-bromann) [reply packages/wdio-sync/webdriverio-core.d.ts]: let me see if I can put this together, definitely makes sense!

**Comment by christian-bromann** [comment]
@mgrybyk fixed the typings for `waitUntil` condition .. it is pretty specific and probably not supported by JSDoc but it works - afaik JSDoc doesn't support such things.

**Pull request merged**